### ISSUE TITLE
Code blocks formats and escapes special characters 

### ISSUE THREAD

**Issue by tarabanton** [RocketChat/Rocket.Chat#6604]
Rocket.Chat Version:  0.54.2
Running Instances: 3
DB Replicaset OpLog: Enabled
Node Version: 4.8.1

Code blocks formats and escapes cpecial characters etc.
Affected, at least, quotes, ampersands, double quotes.

For example:
```
importScripts("https://www.gstatic.com/firebasejs/3.6.8/firebase-app.js?some=true&other=false");
importScripts("https://www.gstatic.com/firebasejs/3.6.8/firebase-messaging.js?some=true&other=false");
```

is converted to

```
importScripts(&quot;https://www.gstatic.com/firebasejs/3.6.8/firebase-app.js?some=true&amp;other=false&quot;);
importScripts("https://www.gstatic.com/firebasejs/3.6.8/firebase-messaging.js?some=true&other=false");
```

**Comment by tarabanton** [comment]
It seems, that this behavior is observed when code block start and\or end sequences` ``` code ``` ` are on the same line as code

**Comment by himbeere** [comment]
Have the same issue. Onliner like this: ``` awk '{print \$2}'); 

leeds to --> screenshot.
![bildschirmfoto 2017-04-05 um 15 02 06](https://cloud.githubusercontent.com/assets/631865/24706694/decc0514-1a10-11e7-9765-84156bc925b4.png)

**Comment by gdelavald** [comment]
This could be related to the implementation made to fix #6311 and changes made to the Regular Expression

**Comment by gdelavald** [comment]
Same as #6592

**Comment by razvanphp** [comment]
We get this too on `0.54.0`, very annoying for developer teams.

**Comment by himbeere** [comment]
"Bug" still in 0.56.0.

**Comment by rabidang3ls** [comment]
I've had this issue, but only if the code block isn't closed correctly. If I leave off the closing ``` then the characters are turned into ampersand character codes, but if the block is closed correctly, the characters remain.

![screenshot 2017-05-30 10 44 46](https://cloud.githubusercontent.com/assets/2701664/26594488/0be24b46-4525-11e7-9a7d-7abc7c8e0eb2.png)

**Comment by martinvonwittich** [comment]
Same as #6789

**Comment by himbeere** [comment]
Seems to be fixed with 0.57.0 here.
